python/vineyard/core/resolver.py

```diff
@@ -230,12 +230,11 @@ def get(
         object_id = client.get_name(name)
 
     # run resolver
-    obj = client.get_object(object_id, fetch=fetch)
-    meta = obj.meta
-    if not meta.islocal and not meta.isglobal:
-        raise ValueError(
-            "Not a local object: for remote object, you can only get its metadata"
-        )
+    if isinstance(client, RPCClient):
+        obj = client.get_object(object_id)
+    else:
+        obj = client.get_object(object_id, fetch=fetch)
```

Comment:
Add two methods `IsIPC()` and `IsRPC()` to the `class ClientBase` in C++ (you can default the return value to `false` and override the corresponding one in `class Client` and `class RPCClient`, respectfully). And expose them to Python bindings as `is_ipc` and `is_rpc` (as `property_readonly`).

Then revise all those `isinstance(...)` checks to use the property.